cmake/pcl_find_cuda.cmake

```diff
@@ -31,25 +22,30 @@ if(CUDA_FOUND)
   # https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#deprecated-features
   
   if(NOT ${CUDA_VERSION_STRING} VERSION_LESS "11.0")
-    set(__cuda_arch_bin "3.5 3.7 5.0 5.2 5.3 6.0 6.1 6.2 7.0 7.2 7.5 8.0")
+    set(__cuda_arch_bin "35;37;50;52;53;60;61;62;70;72;75;80")
   elseif(NOT ${CUDA_VERSION_STRING} VERSION_LESS "10.0")
-    set(__cuda_arch_bin "3.0 3.2 3.5 3.7 5.0 5.2 5.3 6.0 6.1 6.2 7.0 7.2 7.5")
+    set(__cuda_arch_bin "30;32;35;37;50;52;53;60;61;62;70;72;75")
   elseif(NOT ${CUDA_VERSION_STRING} VERSION_LESS "9.0")
-    set(__cuda_arch_bin "3.0 3.2 3.5 3.7 5.0 5.2 5.3 6.0 6.1 6.2 7.0 7.2")
+    set(__cuda_arch_bin "30;32;35;37;50;52;53;60;61;62;70;72")
   endif()
 
-  set(CUDA_ARCH_BIN ${__cuda_arch_bin} CACHE STRING "Specify 'real' GPU architectures to build binaries for, BIN(PTX) format is supported")
-
-  set(CUDA_ARCH_PTX "" CACHE STRING "Specify 'virtual' PTX arch to build PTX intermediate code for. Example: 1.0 1.2 or 10 12")
-  #set(CUDA_ARCH_PTX "1.1 1.2" CACHE STRING "Specify 'virtual' PTX arch to build PTX intermediate code for. Example: 1.0 1.2 or 10 12")
-
-  # Guess this macros will be included in cmake distributive
-  include(${PCL_SOURCE_DIR}/cmake/CudaComputeTargetFlags.cmake)
-  APPEND_TARGET_ARCH_FLAGS()
-
-  # Prevent compilation issues between recent gcc versions and old CUDA versions
-  list(APPEND CUDA_NVCC_FLAGS "-D_FORCE_INLINES")
+  set(CUDA_ARCH_BIN ${__cuda_arch_bin} CACHE STRING "Specify 'real' GPU architectures to build binaries for")
   
-  # Allow calling a constexpr __host__ function from a __device__ function.
-  list(APPEND CUDA_NVCC_FLAGS "--expt-relaxed-constexpr")
+  if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.18")
```

Comment:
There's a `if(POLICY)` or similar construct to check for something wrt a specific policy

Reply:
Its a check if you are using a newer Cmake version, then it sets the policy to new. And sets it to old, if you have a version before 3.18.